Source/PeleLMAdvance.cpp

```diff
@@ -54,6 +60,11 @@ void PeleLM::Advance(int is_initIter) {
    std::unique_ptr<AdvanceAdvData> advData;
    advData.reset(new AdvanceAdvData(finest_level, grids, dmap, m_factory, m_incompressible,
                                     m_nGrowAdv, m_nGrowMAC));
+
+   for (int lev = 0; lev <= finest_level; lev++) {
+     m_extSource[lev]->define(grids[lev], dmap[lev], NVAR, amrex::max(m_nGrowAdv, m_nGrowMAC), MFInfo(), *m_factory[lev]);
```

Comment:
Could m_extSource be smaller than NVAR ? Just trying to minimize how memory we use.

Reply:
And it seems to be defined regardless of whether we use SOOT/SPRAY or none. Is it voluntary ?